### ISSUE TITLE
Regression: network not learning

### ISSUE THREAD

**Issue by aflyax** [dnouri/nolearn#72]
Now that I got nolearn regression to work, it doesn't seem my network is learning. Just wanted to post this here in case someone can identify what's wrong.

``` python
print(np.shape(X)) # (15938, 63)
print(np.shape(y)) # (15938, 1)

# X has been scaled

layers0 = [('input', layers.InputLayer),
           ('dense0', layers.DenseLayer),
           ('dropout1', layers.DropoutLayer),
           ('dense1', layers.DenseLayer),
           ('dropout2', layers.DropoutLayer),
           ('dense2', layers.DenseLayer),
           ('output', layers.DenseLayer)]


net0 = NeuralNet(layers=layers0,
             input_shape=(None, num_features),
             dense0_num_units=200,
             dropout1_p=0.1,
             dense1_num_units=200,
             dropout2_p=0.5,
             dense2_num_units=200,
             
             output_num_units=1,
             output_nonlinearity=identity,
             update_learning_rate=0.01,
             update=nesterov_momentum,
             update_momentum=0.9,
             regression=True,
             eval_size=0.2,
             verbose=1,
             max_epochs=50)

net0.fit(X, y)
```

Output:

```
  input             	(None, 63)          	produces      63 outputs
  dense0            	(None, 200)         	produces     200 outputs
  dropout1          	(None, 200)         	produces     200 outputs
  dense1            	(None, 200)         	produces     200 outputs
  dropout2          	(None, 200)         	produces     200 outputs
  dense2            	(None, 200)         	produces     200 outputs
  output            	(None, 1)           	produces       1 outputs

 Epoch  |  Train loss  |  Valid loss  |  Train / Val  |  Valid acc  |  Dur
--------|--------------|--------------|---------------|-------------|-------
     1  |         nan  |         nan  |          nan  |             |  4.2s
     2  |         nan  |         nan  |          nan  |             |  4.3s
     3  |         nan  |         nan  |          nan  |             |  4.3s

[etc.]
```

I have tried different `output_nonlinearity` options, but all of them produce the same effect. The dataset trains pretty well using gradient boosting classifier, so there is something I am doing wrong obviously.

Thanks.

Update: Weirdly enough, if I make X and y random (same dimensions), it works...

**Comment by dnouri** [comment]
What's the range of values in y?  Maybe scale it to [0, 1].

**Comment by aflyax** [comment]
Thanks! Just doing log(y) worked well!
